src/components/AlertComponent.vue

```diff
@@ -1,14 +1,28 @@
 <template>
-  <div class="alert" :class="'alert-' + type" role="alert">
-    This is a test
+  <div>
+    <div class="alert" :class="'alert-' + type" role="alert">
+      <slot></slot>
+      <button
+        v-if="dismissing"
+        type="button"
+        class="close"
+        data-dismiss="alert"
+        aria-label="Close"
+      >
+        <span aria-hidden="true">&times;</span>
+      </button>
+    </div>
   </div>
 </template>
 
 <script lang="ts">
 import { Component, Vue } from "vue-property-decorator";
 
 @Component({
-  props: ["type"]
+  props: {
+    type: String,
```

Comment:
This is awesome, however to avoid clash of terms please replace type with variant